File: Palindrome.java
class Palindrome{
    public static void main(String [] args){
        int n=1221;
        int b=0;
        int save=n;
        int r;
        while(n!=0){
            r=n%10;
            b=b*10;
            b=b+r;
            n=n/10;
        }
        if( save==b){
            System.out.println("Palindrome");
        }
        else{
            System.out.println("Not a Palindrome");
        }
    }
}

File: Palindrome.c
#include<stdio.h>

void main(){
    int num,sum=0,rev=0,d,x;
    printf("Enter the number:");
    scanf("%d",&num);
    x=num;

    while(num){
        d=num%10;
        num=num/10;
        sum=sum+d;
        rev=rev*10+d;
    }
    
    printf("\nSum of digits : %d",sum);
    printf("\nReverse of the number : %d",rev);
   
    if(x==rev){
        printf("\n The number is Palindrome");
    }else{
        printf("\nThe number is not a Palindrom");
    }
}
